core/block.js

```diff
@@ -985,6 +985,11 @@ Blockly.Block.prototype.appendDummyInput = function(opt_name) {
  * @param {!Object} json Structured data describing the block.
  */
 Blockly.Block.prototype.jsonInit = function(json) {
+  // Add the mutator to the block
+  if (json['mutator'] !== undefined) {
```

Comment:
Push this down to just before the other extensions, so all inputs, etc are initialized before this function is called.